service_management/app/repositories/analysis/dao/AnalysisDao.scala

```diff
@@ -131,6 +146,42 @@ class AnalysisDao @Inject()(
   }
 
   /**
+   * Performs an update action against the DB using the values in the provided
+   * {{{AnalysisEvent}}} argument.
+   *
+   * @param id the EventId associated with the analysis event to update
+   * @param ae the AnalysisEvent to update
+   * @return a result with an option of the updated event
+   */
+  def update(
+      id: EventId,
+      ae: AnalysisEvent
+  ): Future[MusitResult[Option[AnalysisEvent]]] = {
+    val action = updateAction(id, ae).transactionally
```

Comment:
The update in these scenarios are _always_ by id. So the naming it updateByIdAction seems redundant to me.